### ISSUE TITLE
includeBlankOption does not work at select if first value 0

### ISSUE THREAD

**Issue by zonky2** [contao/contao#1022]
**Affected version(s)**
C 4.4.43

**Description**  
its not possible to select `-` - the selected jump to `0`

**How to reproduce**  
see https://github.com/menatwork/contao-multicolumnwizard-bundle/issues/63

**Comment by leofeyer** [comment]
How do I reproduce this?

**Comment by zonky2** [comment]
you must have an option array with `0` as value - see https://github.com/menatwork/contao-multicolumnwizard-bundle/issues/63

to set the attribute selected will be checked for `empty`, instead of empty string

**Comment by fritzmg** [comment]
imho this is a known limitation, as currently the check for an empty value is just something falsey.

**Comment by aschempp** [comment]
I agree with @fritzmg this was always a known limitation to me

**Comment by zonky2** [comment]
Maybe it's time to break the limit... :D

**Comment by aschempp** [comment]
> Maybe it's time to break the limit...

Feel free to provide a solution …

**Comment by zonky2** [comment]
I can try it...

**Comment by zonky2** [comment]
Maybe...
https://github.com/contao/core-bundle/blob/4.4/src/Resources/contao/library/Contao/Widget.php#L1184

```diff
-		return (\is_array($varValues) ? \in_array($strOption, $varValues) : $strOption == $varValues) ? ' checked' : '';
+		return (\is_array($varValues) ? \in_array($strOption, $varValues) : strcmp($strOption, $varValues) === 0) ? ' checked' : '';
```
checked false - fix

```diff
-		return (\is_array($varValues) ? \in_array($strOption, $varValues) : $strOption == $varValues) ? ' selected' : '';
+		return (\is_array($varValues) ? \in_array($strOption, $varValues) : strcmp($strOption, $varValues) === 0) ? ' selected' : '';
```

works at my test case...

**Comment by discordier** [comment]
Shouldn't the `in_array` check become `strict` also?

**Comment by zonky2** [comment]
@discordier you mean `strict = true` ?

**Comment by leofeyer** [comment]
As discussed in Mumble on January 16th, we want to try if `$strOption === (string) $varValues` fixes the issue and does not have any side effects.

**Comment by ausi** [comment]
```php
return \in_array($strOption, array_map('strval', (array) $varValues), true) ? ' checked' : '';
```

**Comment by leofeyer** [comment]
Also, we should check thoroughly if this is really the only place where we need to adjust something.

**Comment by zonky2** [comment]
what are we waiting for?

**Comment by ausi** [comment]
> what are we waiting for?

For someone to fix the bug, check if the fix does not have any side effects and check if this is really the only place where we need to adjust something.

**Comment by zonky2** [comment]
is someone checking or is everyone waiting for someone to do it?

**Comment by m-vo** [comment]
Well, there is no assignee and no PR referencing this issue. And likely it is not that high on everyone's list as this issue hasn't received much attention lately. :smile: But feel free to solve this problem, I'd appreciate it!

**Comment by zonky2** [comment]
done https://github.com/contao/core-bundle/pull/1716

**Comment by ausi** [comment]
Closed in favor of #2350
